src/tests/ftest/util/server_utils.py

```diff
@@ -36,17 +36,54 @@
 import errno
 import yaml
 
+from command_utils import CommandWithParameters
+from command_utils import BasicParameter, FormattedParameter
 from avocado.utils import genio
 
 SESSIONS = {}
 
 DEFAULT_FILE = "src/tests/ftest/data/daos_server_baseline.yaml"
 AVOCADO_FILE = "src/tests/ftest/data/daos_avocado_test.yaml"
 
-
 class ServerFailed(Exception):
     """Server didn't start/stop properly."""
 
+class ServerCommand(CommandWithParameters):
+    """Defines a object representing a server command."""
+
+    def __init__(self, command="daos_server"):
+        """Create a server Command object"""
+
+        super(ServerCommand, self).__init__(command, shell=True, sudo=True)
+
+        self.request = BasicParameter("{}")
+        self.action = BasicParameter("{}")
+        self.targets = FormattedParameter("-t {}")
+        self.config = FormattedParameter("-o {}")
+        self.port = FormattedParameter("-p {}")
+        self.storage = FormattedParameter("-s {}")
+        self.modules = FormattedParameter("-m {}")
+        self.xshelpernr = FormattedParameter("-x {}")
+        self.firstcore = FormattedParameter("-f {}")
+        self.group = FormattedParameter("-g {}")
+        self.attach = FormattedParameter("-a {}")
+        self.sock_dir = FormattedParameter("-d {}")
+    def __str__(self):
+        """Return the command with all of its defined parameters as a string.
+        The daos_server command use the following command structure:
+        daos_server <request> <action> <parameters>
+        Returns:
+            str: the command with all the defined parameters.
+        """
+        params = []
+        for name in self.get_param_names():
+            value = str(getattr(self, name))
+            if (value != "" and name != "request" and name != "action" and
+                    name != "port"):
+                params.append(value)
+        return " ".join([self._command] + params +
+                        [str(getattr(self, "request"))] +
+                        [str(getattr(self, "action"))])
```

Comment:
After this line:
It may make sense for the ServerCommand to have this function as well.  What do you think?:

```
    def prepare(self, path, slots):
        """Prepare the hosts before starting daos server.

        Args:
            path (str): location to write the hostfile
            slots (int): slots per host to use in the hostfile

        Raises:
            ServerFailed: if there is any errors preparing the hosts

        """
        # Kill any doas servers running on the hosts
        kill_server(self.hosts)

        # Clean up any files that exist on the hosts
        clean_server(self.hosts)

        # Ensure the environment for the daos server on each host
        okay, failed, file = node_setup_okay(self.hosts, NodeListType.SERVER)
        if not okay:
            raise ServerFailed(
                "Server node {} does not have directory {} set up correctly "
                "for user {}.".format(failed, file, getpass.getuser()))

        # Create the hostfile
        self.hostfile = write_host_file(self.hosts, path, slots)
```

Reply:
I agree, looks good and it's a necessary check.